### PR TITLE
feat: login component and logic added

### PR DESCRIPTION
**Pull request by saifudinbadani**

### CODE REVIEW

**Review by MurtazaBagwala07 — APPROVED** [review]
Code looks good
Good to merge

File: src/App.js [hunk 1]
```diff
@@ -5,17 +5,19 @@ import { HomePage } from "./components/HomePage/HomePage";
...
       <Routes>
         <Route path="/" element={<LandingPage />} />
-        <Route path="/homePage" element={<HomePage />} />
-        <Route path="/archive" element={<ArchivePage />} />
-        <Route path='/login' element={<LoginPage />} />
-        <Route path='*' element={<PageNotFound />} />
+        <Route path="/homePage" element={isLoggedIn && <HomePage />} />
+        <Route path="/archive" element={isLoggedIn && <ArchivePage />} />
+        <Route path='/login' element={!isLoggedIn && <LoginPage />} />
+        <Route path='*' element={ <PageNotFound />} />
```
Comment (devsurajrai) [inline src/App.js:20]: Try to make a "RequiresAuth" for protected routes

File: src/components/LoginPage/Login.js [hunk 2]
```diff
@@ -3,35 +3,63 @@ import { Link } from 'react-router-dom';
...
+    }
+
+    const loginHandler = async () => {
+        const respData = await loginService(loginData.email, loginData.password)
+        if(respData){
+            setInitialAuth({isLoggedIn: true,
+                token: respData.data.encodedToken});
+            localStorage.setItem('token', respData.data.encodedToken)
+            localStorage.setItem('isLoggedIn', true)
+            navigate('/homePage')
```
Comment (devsurajrai) [inline src/components/LoginPage/Login.js:27]: i think there is no need to store the logged in state in local storage. If token is present in the local storage you can set the logged in state to true

**Review by devsurajrai — COMMENTED** [review]
Everything looks good. Just try to apply the suggestion provided.

**Pull request merged**